### ISSUE TITLE
Indicators give errors for keys not in action keymap

### ISSUE THREAD

**Issue by roshanshariff** [oantolin/embark#369]
Currently, `embark-verbose-indicator` and `embark-which-key-indicator` (from the wiki) give errors when typing a key sequence that is not in the action keymap, but is bound globally. This is because they show the keymap returned by `(lookup-key keymap prefix)`, which is nil when the prefix is not bound in the keymap. Here is an updated version of `embark-which-key-indicator` that solves the problem:
```emacs-lisp
(defun embark-which-key-indicator ()
  "An embark indicator that displays keymaps using which-key.
The which-key help message will show the type and value of the
current target followed by an ellipsis if there are further
targets."
  (lambda (&optional keymap targets prefix)
    (if (null keymap)
        (which-key--hide-popup-ignore-command)
      (when-let ((prefix-keymap
                  (if prefix
                      (let ((overriding-terminal-local-map keymap))
                        (key-binding prefix 'accept-default))
                    keymap))
                 ((keymapp prefix-keymap)))
        (which-key--show-keymap
         (if (eq (caar targets) 'embark-become)
             "Become"
           (format "Act on %s '%s'%s"
                   (plist-get (car targets) :type)
                   (embark--truncate-target (plist-get (car targets) :target))
                   (if (cdr targets) "…" "")))
         prefix-keymap nil nil t)))))
```
A similar change can be made to `embark-verbose-indicator` as well, though perhaps the user should be able to decide whether the indicator remains visible once an unbound prefix is entered.

If you agree with this change, I can prepare a pull request fixing these functions (and adding the customization variable).

**Comment by minad** [comment]
> A similar change can be made to |embark-verbose-indicator| as well,
> though perhaps the user should be able to decide whether the indicator
> remains visible once an unbound prefix is entered.
> 
> If you agree with this change, I can prepare a pull request fixing these
> functions (and adding the customization variable).

I don't see an error for the embark-verbose-indicator. The indicator
stays alive but is updated and shows nothing (the empty/nil keymap). It
would be better if we lookup the prefix in the currently active keymaps.
This can be achieved by using

(or (lookup-key keymap prefix) (key-binding prefix))

instead of only lookup-key.

**Comment by roshanshariff** [comment]
Right, the error happens if you type a key sequence that is bound to a prefix keymap. So for example `C-x r` shows "Error running timer: (wrong-type-argument keymapp 1)" in the echo area. From looking at the documentation of `lookup-key` it appears this is happening:
1. When the prefix is `C-x`, it returns nil, which doesn't seem to cause an error.
2. When it is `C-x r` it returns the number 1, which is the number of keys in the prefix it took to reach a non-prefix key.

So looking at your pull request #370, you also have to check whether the result of `lookup-key` is a number. Also, it's probably good to use `(key-binding 'accept-default)` to more faithfully replicate how `read-key-sequence` deals with default keybindings; maybe this should also be changed wherever `key-binding` is used in embark.el. Thanks @minad for your very quick response and help with this bug!

**Comment by minad** [comment]
> Right, the error happens if you type a key sequence that is bound to a
> prefix keymap. So for example |C-x r| shows "Error running timer:
> (wrong-type-argument keymapp 1)" in the echo area.
> So looking at your pull request #370
> <https://github.com/oantolin/embark/pull/370>, you also have to check
> whether the result of |lookup-key| is a number.

Right, I wasn't aware that lookup-key can return numbers. I fixed this
in #370.

> Also, it's probably good
> to use |(key-binding 'accept-default)| to more faithfully replicate how
> |read-key-sequence| deals with default keybindings; maybe this should
> also be changed wherever |key-binding| is used in embark.el.

Both lookup-key and key-binding accept this accept-default argument. I
added the argument in #370 where I think it makes sense.

**Comment by roshanshariff** [comment]
Excellent! I can confirm that #370 solves the issue for me. I've also updated `embark-which-key-indicator` in the [wiki](https://github.com/oantolin/embark/wiki/Additional-Configuration#use-which-key-like-a-key-menu-prompt). Maybe for discoverability it should also be updated in the readme?

Thanks again for all your help!
